spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -216,18 +260,103 @@ def execute(
         conditional_print(message=message, verbose=self.verbose)
         return status, message
 
+    def pre_execution_rl_loop(self, goal_dict: Dict[str, Any]):
+        """
+        Updates (resets) the logged data at the start of the navigation execution loop
+
+        Args:
+            goal_dict: Dict[str, Any] containing necessary goal information for navigation
+                                      Required key: "nav_target" (x,y,theta) where robot needs to navigate to
+        """
+        nav_target = goal_dict.get("nav_target")
+        self.env.say(f"Navigating to {nav_target}")
+
+        # List of Dicts to store trajectory for nav_target (this also clears the previous data)
+        self.robot_trajectory = []
+
+        # Set start time for recording before starting navigation (this also resets the previous start time data)
+        self.start_time = time.time()
 
-class Pick:
+    def in_execution_rl_loop(self, goal_dict: Dict[str, Any]):
+        """
+        Updates the logged data at every step of the navigation execution loop
+
+        Args:
+            goal_dict: Dict[str, Any] containing necessary goal information for navigation
+                                      Required key: "nav_target" (x,y,theta) where robot needs to navigate to
+        """
+        # Record trajectories at every step
+        self.robot_trajectory.append(
+            {
+                "timestamp": time.time() - self.start_time,
+                "pose": [
+                    self.env.x,
+                    self.env.y,
+                    np.rad2deg(self.env.yaw),
+                ],
+            }
+        )
```

Comment:
I believe in the next iteration we may be able to collapse all trajectory-logging into one function, right? Where we just store the entire joint-position of arm + position of base for every skill. We would then need to add a `read_nav/pick/place_trajectory` to then filter out this to return obs-space expected by each skill, but that should be easy.

Reply:
Yes, something like that should come following this PR. Exact procedure for how to achieve that isn't concrete yet.

Reply:
We are currently logging base. Can add arm states as a part of next PR